### PR TITLE
Moving `value($X)` information to experiments

### PR DESCRIPTION
**Pull request by adamkvitek**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

File: docs/writing-rules/pattern-syntax.mdx [hunk 1]
```diff
@@ -492,34 +492,23 @@ Creating a buffer using size
...
-
-To retrieve the correct value in the case described above, use `value($X)` in the rule message (for example (`Creating a buffer using value($X)`). Semgrep replaces the `value($X)` with the underlying propagated value of the metavariable `$X` if it computes one (otherwise, Semgrep uses the matched value).
-
-Run the following example in Semgrep Playground to see the message (click **Open in Editor**, and then **Run**, unroll the **1 Match** to see the message):
-
-<iframe title="Metavariable value in message example" src="https://semgrep.dev/embed/editor?snippet=returntocorp:value-in-message-example" width="100%" height="432" frameborder="0"></iframe>
+:::info
+There are specific limitations for nested variables which Semgrep cannot process correctly under specific circumstances. For more information see documentation of experimental feature [Displaying propagated value of metavariable](/experiments/display-propagated-metavariable-message)
```
Comment (adamkvitek) [inline docs/writing-rules/pattern-syntax.mdx:510]: Can we call them "nested variables"? @emjin

Comment (emjin) [reply docs/writing-rules/pattern-syntax.mdx]: No, that's not really right. Also I wouldn't really call this a "limitation". It makes it sound like sometimes we don't have a matched value. Maybe we can say "If you're using Semgrep's advanced dataflow features, you might want to access more information than is present now. etc."

Comment (adamkvitek) [reply docs/writing-rules/pattern-syntax.mdx]: Ok, changing to: If you're using Semgrep's advanced dataflow features, see documentation of experimental feature [Displaying propagated value of metavariable](/experiments/display-propagated-metavariable-message).

File: docs/writing-rules/pattern-syntax.mdx [hunk 2]
```diff
@@ -465,61 +465,46 @@ match:
...
-Testing code:
+1. Find the metavariable used in Semgrep rule. For example:
+    ```yaml
+    - pattern: $MODEL.set_password(…)
+    ```
+    This formula uses `$MODEL` as a metavariable.
+2. Insert the metavariable to rule message:
+    ```yaml
+    - message: Setting a password on $MODEL
+    ```
```
Comment (emjin) [inline docs/writing-rules/pattern-syntax.mdx:503]: Can we give them a way to try this out? Or at least indicate what the result would be?

**Review by emjin — APPROVED** [review]
Really happy with this layout now! Left one comment, otherwise looks good

File: docs/writing-rules/pattern-syntax.mdx (hunk 2 above)
Comment (adamkvitek) [reply docs/writing-rules/pattern-syntax.mdx:503]: @emjin of course! Please, can you give me appropriate test code? 🙏

Comment (adamkvitek) [reply docs/writing-rules/pattern-syntax.mdx:482]: Done))))

**Pull request merged**